viewer/test/parser.test.ts

```diff
@@ -0,0 +1,63 @@
+import * as expect from "expect";
+import { Parser } from "../src/parser";
+import * as fs from "fs-extra";
+import { expectedLocations, expectedControlPointTransition, expectedControlPointTriggerable, expectedStateMachines, expectedStates, expectedTransitionPatternStates, expectedLinkDataArray, expectedFinalState, componentName, entryPointStateMachine } from "./parser.expectation";
+
+describe("Test Parser", () => {
+
+    let parser: Parser;
+    let componentGraphicalModel, model, graphical;
+    beforeEach(() => {
+        model = fs.readFileSync("./test/WorldHello.cxml").toString();
+        graphical = fs.readFileSync("./test/WorldHello_Graphical.xml").toString();
+        parser = new Parser({ model, graphical });
+        parser.promisifyParserString()
+    });
+
+    test("Test getLocationState method: should get the right locations of the state", (done) => {
+        parser.promisifyParserString(graphical).then((graphicalJson) => {
```

Comment:
I don't understand the way it should work. It's so complicated!
What I was expecting: parser.parse(...).then(...)